### PR TITLE
Adapt to https://github.com/coq/coq/pull/19530

### PR DESCRIPTION
**Pull request by proux01**
Adapt to https://github.com/coq/coq/pull/19530
This should be backward compatible and can be merged at your convenience, before the upstream PR.

### CODE REVIEW

File: src/Rewriter/Rewriter/Reify.v [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ Module Compilers.
                   debug_Constr_check (fun () => mkApp '@cons [listT; rlem; rlems])
              | nil => debug_Constr_check (fun () => mkApp '@nil [listT])
              | _
-               => let list_map := (eval cbv delta [List.map] in '(@List.map)) in
+               => let list_map := (eval cbv delta [ListDef.map] in '(@List.map)) in
```
Comment (JasonGross) [inline src/Rewriter/Rewriter/Reify.v:1027]: This change is not backwards compatible.  Why is it here?

File: src/Rewriter/Util/Bool/Reflect.v [hunk 2]
```diff
@@ -59,8 +59,8 @@ Proof. apply reflect_to_brel; assumption. Qed.
 Lemma reflect_rect_dep {P b} (Q : reflect P b -> Type)
       (H : forall pf : if b then P else ~P,
           (if b return (reflect P b -> Type) -> (if b then P else ~P) -> Type
-           then fun Q pf => Q (ReflectT _ pf)
-           else fun Q pf => Q (ReflectF _ pf))
+           then fun Q pf => Q (@ReflectT _ pf)
+           else fun Q pf => Q (@ReflectF _ pf))
```
Comment (JasonGross) [inline src/Rewriter/Util/Bool/Reflect.v:63]: Why is this necessary?

File: src/Rewriter/Rewriter/Reify.v (hunk 1 above)
Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v:1027]: Unfortunately the list in eval doesn't resolves abbreviations so I guess there is no solution short of just putting `map` here if we want backward compatibility.

File: src/Rewriter/Rewriter/Reify.v [hunk 3]
```diff
@@ -1454,7 +1454,7 @@ Module Compilers.
                                          => fun var => myapp' _ (@interp_rewrite_rules' var) (lems' var)
                                        end)) in
                   let len := lazy_match! (eval cbv in constr:(match $interp_rewrite_rules return _ with interp_rewrite_rules' => fun var => List.length (interp_rewrite_rules' var) end)) with (fun _ => ?n) => n end in
-                  let adjusted_specs := (eval cbv [Datatypes.app List.repeat] in
+                  let adjusted_specs := (eval cbv [Datatypes.app ListDef.repeat] in
```
[the remark this answers is not in the document]
Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v:1457]: same problem

File: src/Rewriter/Rewriter/Reify.v (hunk 1 above)
Comment (JasonGross) [reply src/Rewriter/Rewriter/Reify.v]: Is `List.map` an abbreviation now?  Or do you mean it doesn't peek under `Include`?  But it should be the same constant in `List.map` and `'(@List.map)`, no?  Is the definition of `List` changing?

Comment (JasonGross) [reply src/Rewriter/Rewriter/Reify.v]: And can we fix this deficiency in `eval` lists?

Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v]: I'm trying that.

Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v]: Funily enough, it's an Ltac2 specific issue, Ltac1 handles it fine

Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v]: Fixed

**Comment by proux01** [comment]
@JasonGross all the points you raised are now adressed and this should be backward compatible.

File: src/Rewriter/Rewriter/Reify.v (hunk 1 above)
Comment (JasonGross) [reply src/Rewriter/Rewriter/Reify.v]: What was the Ltac2 issue? / What was the fix? / Was there a Coq PR fixing Ltac2 here?

Comment (proux01) [reply src/Rewriter/Rewriter/Reify.v]: It was a one liner https://github.com/coq/coq/pull/19530/commits/86b6ece910fe914c5890a7587e6a859d81150771

**Comment by proux01** [comment]
Thanks
